pkg/yttlibrary/assert.go

```diff
@@ -25,6 +28,69 @@ var (
 
 type assertModule struct{}
 
+func (b assertModule) Equals(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if args.Len() != 2 {
+		return starlark.None, fmt.Errorf("expected two arguments")
+	}
+
+	expected := args.Index(0)
+	actual := args.Index(1)
+	if _, notOk := expected.(starlark.Callable); notOk {
+		return starlark.None, fmt.Errorf("expected argument not to be a function, but was %T", expected)
+	}
+	if _, notOk := actual.(starlark.Callable); notOk {
+		return starlark.None, fmt.Errorf("expected argument not to be a function, but was %T", actual)
+	}
+
+	var expectedYamlString string
+	var actualYamlString string
+
+	if expected.Type() == "yamlfragment" {
+		expectedStarlarkValue, err := core.NewStarlarkValue(expected).AsGoValue()
+		if err != nil {
+			return starlark.None, err
+		}
+		document := yamlmeta.Document{Value: expectedStarlarkValue}
+		expectedYaml, err := document.AsYAMLBytes()
+		if err != nil {
+			return starlark.None, err
+		}
+		expectedYamlString = string(expectedYaml)
+	}
+
+	if actual.Type() == "yamlfragment" {
+		actualStarlarkValue, err := core.NewStarlarkValue(actual).AsGoValue()
+		if err != nil {
+			return starlark.None, err
+		}
+		document := yamlmeta.Document{Value: actualStarlarkValue}
+		actualYaml, err := document.AsYAMLBytes()
+		if err != nil {
+			return starlark.None, err
+		}
+		actualYamlString = string(actualYaml)
+	}
+
+	if expected.Type() == "yamlfragment" && actual.Type() != "yamlfragment" {
+		return starlark.None, fmt.Errorf("Not equal:\n"+
+			"expected:\n%s\nactual:\n%s", expectedYamlString, actual.String())
+	} else if expected.Type() != "yamlfragment" && actual.Type() == "yamlfragment" {
+		return starlark.None, fmt.Errorf("Not equal:\n"+
+			"expected:\n%s\n\nactual:\n%s", expected.String(), actualYamlString)
+	} else if expected.Type() == "yamlfragment" && actual.Type() == "yamlfragment" {
+		if expectedYamlString != actualYamlString {
+			return starlark.None, fmt.Errorf("Not equal:\n"+
+				"expected:\n%s\nactual:\n%s", expectedYamlString, actualYamlString)
+		}
+	} else {
+		if !reflect.DeepEqual(expected, actual) {
+			return starlark.None, fmt.Errorf("%s is not equal to the expected value %s", actual.String(), expected.String())
+		}
```

Comment:
This error message format is different than the cases above. Is the difference making a difference?

If so, good stuff.  If not, let's make them the same.

When there are options to simplify (without going so far as to sacrifice other qualities), it's another fine way to avoid bugs.